src/core/createConfig.js

```diff
@@ -48,18 +51,46 @@ const createConfig = config => {
       return keys;
     },
     /**
-     * Adds schema information to the existing configuration schema.
+     * Adds more validators to any existing validators.
      */
-    extendSchema: schemaAddition => {
-      assign(cfg.schema, schemaAddition);
-      return cfg.schema;
+    addValidators: validators => {
+      assign(cfg.validators, validators);
+      return cfg.validators;
     },
     /**
-     * Validates the configuration against the defined schema.
+     * Validates the configuration against the defined validators.
      */
     validate: () => {
-      // TODO: Validate existing configuration against defined schema.
-      return true;
+      const keys = Object.keys(cfg.validators);
+      const errors = keys.reduce((ac, key) => {
+        const validator = cfg.validators[key];
+        const currentValue = cfg.get(key);
+        if (
+          !validator.validate &&
+          currentValue == null &&
+          validator.defaultValue
```

Comment:
This should probably be `validator.hasOwnProperty('defaultValue')` to account for default values of `0`, empty string, etc.